### PR TITLE
[Toolchain] Fix SSR and react hooks support

### PR DESCRIPTION
**Pull request by damassi**
This fixes a few things: 
- The pesky styled-components SSR issue where a linked module would break server rendering due to the requirement that a single static context be present. We ran into this issue with [react hooks](https://github.com/facebook/react/issues/13991) earlier today during SSR so that's been added to the list of aliased modules.
- Once the SSR pass was fixed client-side errors began to appear oddly enough, which I was able to track down to [react-hot-loader](https://github.com/gaearon/react-hot-loader/issues/1088#issuecomment-446938770). 

With these items fixed I was able to remove the `yarn start:ssr` command and have tested cross-linking all the way across from reaction to palette.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -27,6 +27,20 @@ if (NODE_ENV === "development") {
...
+// Force resolution of potentially `yarn link`'d modules to the local node_modules
+// folder. This gets around SSR issues involving single react context requirements.
+// This is server-side only. Client-side must be resolved via webpack.
+setAliases({
+  react: path.resolve(path.join(__dirname, "../node_modules/react")),
+  "react-dom": path.resolve(path.join(__dirname, "../node_modules/react-dom")),
+  "styled-components": path.resolve(
+    path.join(__dirname, "../node_modules/styled-components")
+  ),
+})
```
Comment (ds300) [inline src/index.js]: 👌

**Comment by damassi** [comment]
> As for the map of aliases, is there any way we can extract them from here and the web pack config and maintain a single authoritative list that both places can import?

We can do that, something like a `sharedAliases` file. On the webpack side we're adding a few other aliases but those are old jQuery client libs, so we can cleanly separate.

**Comment by alloy** [comment]
Ace 👍 Feel free to either merge and iterate or add it now.

**Comment by damassi** [comment]
Looked at the [webpack side of things](https://github.com/damassi/force/blob/master/webpack/envs/baseConfig.js#L77-L78) and the form required is different enough to warrant a bit of duplication. That said, I added a comment on the Webpack side and we can return if this list grows a bit.

**Pull request merged**